### PR TITLE
Add --insert-license-after-regex option

### PR DESCRIPTION
**Pull request by mdeweerd**
Allows to implement #38 .

`--insert-license-after-regex '^<\?php$'` will insert the licence after the first appearing '<?php' or at the end of the file.

### CODE REVIEW

File: pre_commit_hooks/insert_license.py [hunk 1]
```diff
@@ -194,9 +200,15 @@ def license_not_found(remove_header, license_info, src_file_content, src_filepat
         index = 0
         for line in src_file_content:
             stripped_line = line.strip()
-            # Special treatment for shebang, encoding and empty lines when at the beginning of the file
+            # Special treatment for user provided regex,
+            # or shebang, file encoding directive,
+            # and empty lines when at the beginning of the file.
             # (adds license only after those)
-            if stripped_line.startswith("#!") \
+            if after_regex is not None and after_regex != "":
```
Comment (Lucas-C) [inline pre_commit_hooks/insert_license.py:207]: A simple `if after_regex:` would cover both cases of `None` and the empty string

File: pre_commit_hooks/insert_license.py [hunk 2]
```diff
@@ -194,9 +200,15 @@ def license_not_found(remove_header, license_info, src_file_content, src_filepat
...
+                if re.match(after_regex, stripped_line):
+                    index += 1  # Skip matched line
+                    break       # And insert after that line.
+            elif stripped_line.startswith("#!") \
```
Comment (Lucas-C) [inline pre_commit_hooks/insert_license.py:211]: That means that if `--after-regex` is provided, shebang & `# -*- coding` lines are not skipped?
Could you please add a test file to check for this scenario?

File: tests/insert_license_test.py [hunk 3]
```diff
@@ -48,6 +48,7 @@
             ('main_iso8859_without_license.cpp', '/*|\t| */', 'main_iso8859_with_license.cpp', True, None),
             ('module_without_license.txt', '', 'module_with_license_noprefix.txt', True, None),
             ('module_without_license.py', '#', 'module_with_license_nospace.py', True, ['--no-space-in-comment-prefix']),
+            ('module_without_license.php', '/*| *| */', 'module_with_license.php', True, ['--insert-license-after-regex', '^<\\?php$']),
```
Comment (Lucas-C) [inline tests/insert_license_test.py:51]: 👍

**Comment by Lucas-C** [comment]
@mdeweerd Could you please rebase this PR?
It currently includes many commits that are already in the `master` branch,
making it difficult to visualize the effective changes included in this PR.

**Comment by mdeweerd** [comment]
I tried to rebase, but I possibly need to request a new push.
It looks fine on https://github.com/Lucas-C/pre-commit-hooks/compare/master...mdeweerd:pre-commit-hooks:insert_after , but not when diffing here.

**Pull request closed without merging**